File: 0053-maximum-subarray/0053-maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int sum = 0,maxsum=nums[0];


    for(int i=0;i<numsSize;i++){
        if(sum<0){
            sum = 0;
        }
        sum+=nums[i];

        if(sum>maxsum) maxsum = sum;
    }   
    return maxsum;
}

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum =nums[0];
        int currsum = 0;

        for(int i : nums){
            if(currsum < 0) currsum = 0;
            currsum += i;
            System.out.println(currsum);
            if(currsum > maxSum ) maxSum = currsum;
        }
        return maxSum;
    }
}